client/src/components/AddItem.js

```diff
@@ -54,9 +55,40 @@ const useStyles = makeStyles((theme) => ({
 
 const demoListsArray = ["Clothes", "Furniture", "Luxury"];
 
+const getItem = async (input) => {
+  const response = await fetch("/scrape", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ URL: input }),
+  });
+  return response.json();
+};
+
+const emptyItem = {
+  shopURL: null,
+  title: null,
+  oldPrice: null,
+  price: null,
+  imgURL: null,
+};
+
 const AddItem = () => {
-  const [selectedItem, setSelectedItem] = useState("none");
+  const [inputLink, setInputLink] = useState("");
+  const [item, setItem] = useState(emptyItem);
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [selectedList, setSelectedList] = useState("none");
   const classes = useStyles();
+  const openPopup = () => setPopupOpen(true);
```

Comment:
let's make openPopup() be `changePopup = () => setPopupOpen(!popupOpen)`. Then we can pass changePopup (or better name of your choosing) down to AddItemPopup. It's best practice in React not to call a setState function from outside the component where that state lives.

Reply:
Understood, however the Modal component has an `onClose` parameter rather than `onChange`.

Would it be best instead to pass a closePopup function that is:

`closePopup = () => setPopupOpen(false)`

down to AddItemPopup?

That way it's concise in its functionality, but then also the setState won't leave the parent component.

Reply:
Ok combining your other suggestion below I've changed AddItem to instead pass down:
```
const closePopup = () => {
    setPopupOpen(false);
    setItem({});
};
```

to AddItemPopup.

Reply:
Technically we can pass an onClose any function we want, but given that we're also clearing the setItem on close this is cleaner!